### PR TITLE
chore: add better error messages

### PR DESCRIPTION
**Pull request by tzneal**
**Description**

Detect some typos and try to clarify the error message that occurs when a pod requires a label that we don't know about.

**How was this change tested?**

Unit testing & deployed to EKS.

Samples of the improved error messages are in the unit test.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/scheduling/requirements.go [hunk 1]
```diff
@@ -120,12 +120,65 @@ func (r Requirements) Compatible(requirements Requirements) (errs error) {
...
 		}
-		errs = multierr.Append(errs, fmt.Errorf("key %s does not have known values", key))
+		errs = multierr.Append(errs, fmt.Errorf("label %q does not have known values%s", key, labelHint(r, key)))
 	}
 	// Well Known Labels must intersect, but if not defined, are allowed.
 	return multierr.Append(errs, r.Intersects(requirements))
 }
 
+// editDistance is an implementation of edit distance from Algorithms/DPV
+func editDistance(s, t string) int {
```
Comment (ellistarn) [inline pkg/scheduling/requirements.go:130]: Thoughts on moving to functional?

**Review by ellistarn — APPROVED** [review]
Neato!

File: pkg/scheduling/requirements.go (hunk 1 above)
Comment (tzneal) [reply pkg/scheduling/requirements.go:130]: Sure if we have a second use.  Any ideas for where we might also use it?

Comment (tzneal) [reply pkg/scheduling/requirements.go:130]: Just going to resolve and merge for now, happy to move later if we get another use case.

**Pull request merged**